### DISCUSSION TITLE
Prevent running multiple instances

### DISCUSSION THREAD

**Discussion by max-baz** [isamert/scli#171] [Ideas]
Right now running multiple instances of `scli` will not work anyway (one of them will be locked on `signal-cli`), and by accidentally doing this I somehow managed to wipe a few days of history (opened second instance, realized mistake and closed but closed both).

We could perhaps just add some code to prevent history overwrite until we get a lock from `signal-cli`, but maybe there is more we could do, have some sort of our own lock, and quicker abort if it's a second instance?

**Comment by exquo** [comment]
Yes, this should be address. Thanks for raising it!

Multiple instances probably shouldn't be allowed even if both run without `--save-history`, as they can still call signal-cli's `dbus` methods (even though only one signal-cli's `daemon` instance will be running), showing different chat histories, and perhaps other unpredictable behavior. 

Multiple instances should be allowed, however, if they use different `--username`s (accounts / phone numbers). But #172 will need to be fixed first.
